src/lava/proc/sparse/models.py

```diff
@@ -0,0 +1,359 @@
+        weight_delay_from_1 = vstack([weights.multiply(delays == k)
+                                      for k in range(1, np.max(delays) + 1)])
+        # Create weight matrix at delays == 0
+        r, c, _ = find(delays)
+        weight_delay_zeros = weights.copy()
+        weight_delay_zeros[r, c] = 0
+        weight_delay_zeros.eliminate_zeros()
+        return vstack([weight_delay_zeros, weight_delay_from_1])
+
+    def calc_act(self, s_in) -> np.ndarray:
+        """
+        Calculate the activations by performing del_wgts * s_in. This matrix
+        is then summed across each row to get the activations to the output
+        neurons for different delays. This activation vector is reshaped to a
+        matrix of the form
+        (n_flat_output_neurons * (max_delay + 1), n_flat_output_neurons)
+        which is then transposed to get the activation matrix.
+        """
+        return np.reshape(self.get_del_wgts(self.weights,
+                                            self.delays).dot(s_in),
+                          (np.max(self.delays) + 1,
+                          self.weights.shape[0])).T
+
+    def update_act(self, s_in):
+        """
+        Updates the activations for the connection.
+        Clears first column of a_buff and rolls them to the last column.
+        Finally, calculates the activations for the current time step and adds
+        them to a_buff.
+        This order of operations ensures that delays of 0 correspond to
+        the next time step.
+        """
+        self.a_buff[:, 0] = 0
+        self.a_buff = np.roll(self.a_buff, -1)
+        self.a_buff += self.calc_act(s_in)
+
+
+@implements(proc=DelaySparse, protocol=LoihiProtocol)
+@requires(CPU)
+@tag("floating_pt")
+class PyDelaySparseModelFloat(AbstractPyDelaySparseModel):
+    """Implementation of Conn Process with Sparse synaptic connections in
+    floating point precision. This short and simple ProcessModel can be used
+    for quick algorithmic prototyping, without engaging with the nuances of a
+    fixed point implementation. DelaySparse incorporates delays into the Conn
+    Process.
+    """
+    s_in: PyInPort = LavaPyType(PyInPort.VEC_DENSE, bool, precision=1)
+    a_out: PyOutPort = LavaPyType(PyOutPort.VEC_DENSE, float)
+    a_buff: np.ndarray = LavaPyType(np.ndarray, float)
+    # weights is a 2D matrix of form (num_flat_output_neurons,
+    # num_flat_input_neurons) in C-order (row major).
+    weights: np.ndarray = LavaPyType(csr_matrix, float)
+    # delays is a 2D matrix of form (num_flat_output_neurons,
+    # num_flat_input_neurons) in C-order (row major).
+    delays: np.ndarray = LavaPyType(csr_matrix, int)
+    num_message_bits: np.ndarray = LavaPyType(np.ndarray, int, precision=5)
+
+    def run_spk(self):
+        # The a_out sent on a each timestep is a buffered value from dendritic
+        # accumulation at timestep t-1. This prevents deadlocking in
```

Comment:
Accumulation at time step t-1; this prevents..

Reply:
done